main.c

```diff
@@ -237,6 +242,29 @@ int main(int argc, char *argv[]) {
 
 			free(geometry_str);
 			break;
+		case 't':
+			if (strcmp(optarg, "jpeg") == 0) {
+				output_filetype = strdup(optarg);
+			} else if (strcmp(optarg, "png") != 0) {
+				fprintf(stderr, "invalid filetype [png/jpeg]\n");
+				return EXIT_FAILURE;
+			}
+			break;
+		case 'q':
+			if (strcmp(output_filetype, "jpeg") != 0) {
+				fprintf(stderr, "quality is used only for jpeg filetype\n");
+				return EXIT_FAILURE;
+			} else {
+				if (sscanf(optarg, "%i", &jpeg_quality) != 1) {
```

Comment:
I prefer to use `strtol` because it makes it easier to check for errors (`*endptr == '\0'`).